azurerm/internal/services/netapp/netapp_volume_resource.go

```diff
@@ -390,7 +390,7 @@ func resourceNetAppVolumeRead(d *schema.ResourceData, meta interface{}) error {
 		if err := d.Set("mount_ip_addresses", flattenNetAppVolumeMountIPAddresses(props.MountTargets)); err != nil {
 			return fmt.Errorf("setting `mount_ip_addresses`: %+v", err)
 		}
-		if props.DataProtection.Replication != nil {
+		if props.DataProtection != nil && props.DataProtection.Replication != nil {
 			if err := d.Set("data_protection_replication", flattenNetAppVolumeDataProtectionReplication(props.DataProtection)); err != nil {
```

Comment:
from Terraform's side we should be setting this field in all cases, since the flatten function returns an empty slice when the `if` conditions above aren't met - so could we remove the if statement:

```suggestion
		if err := d.Set("data_protection_replication", flattenNetAppVolumeDataProtectionReplication(props.DataProtection)); err != nil {
```